lib/GuzzleAdapter.php

```diff
@@ -40,7 +40,26 @@ public function __construct(ClientInterface $client)
     {
         $this->client = $client;
     }
-
+    
+    /**
+     * Submit an HTTP HEAD request
+     *
+     * @param UriInterface $url The url to send the request to $uri
+     * @param array $headers A key/value pair array of headers to send with the request
+     *
+     * @return ResponseInterface The HTTP response (PSR implementation)
+     *
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function head(UriInterface $url, array $headers = [])
```

Comment:
```suggestion
    public function head(UriInterface $url, array $headers = []): ResponseInterface
```
@joelbitar is return type hinting supported in all php versions we must support?

Reply:
@nunocosta-divido @andivido @joelbitar I think this is will be OK? PHP5 supports classes for “type declarations” and “return type declarations”?

https://www.brainbell.com/php/type-declarations-hints.html

Reply:
I'm pretty sure you can not do this in PHP5, needs to be PHP7